src/System.Management.Automation/engine/parser/PSType.cs

```diff
@@ -467,12 +467,23 @@ private bool ShouldImplementProperty(string name, Type type)
                     {
                         foreach (var property in interfaceType.GetProperties())
                         {
-                            _interfaceProperties.Add(Tuple.Create(property.Name, property.PropertyType));
+                            _abstractProperties.Add(Tuple.Create(property.Name, property.PropertyType));
+                        }
+                    }
+
+                    if (_typeBuilder.BaseType.IsAbstract)
+                    {
+                        foreach (var property in _typeBuilder.BaseType.GetProperties())
```

Comment:
I believe this will need to include `BindingFlags.NonPublic` as `protected` members should still be found right? Likewise `GetAccessors` below likely needs `nonPublic: true`.

Then a check like

```cs
accessor.IsFamilyOrAssembly || accessor.IsFamily || assessor.IsPublic
```

Interfaces may also actually need that check too since C# 8 added the ability to mark members as protected. Though that wouldn't be fixing a regression so likely fine to leave out of this PR.

Reply:
But we don't create properties or methods with the "protected" (`family`) modifier, and we don't have a keyword in PowerShell language to declare a property as `protected`, we only create `public` (by default or with the `hidden` keyword) and `static` (`static` keyword) members.

When a base type (`MyBase`) has a protected abstract property `Name`, the class definition `class MySub : MyBase { [string]$Name }` will fail on 7.2 and 7.3 as well today:

**7.2.18**
![image](https://github.com/PowerShell/PowerShell/assets/127450/0d374704-f8fa-4986-ac9f-551330962cf4)

**7.3.11**
![image](https://github.com/PowerShell/PowerShell/assets/127450/049513a7-9d27-4765-a501-32fb3024ab2e)